File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> vals;
        for (int i = 0; i < nums.size(); ++i) {
            vals.insert(nums[i]);
        }
        if (vals.size() != nums.size()) {
            return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        int longest = 1;
        if (nums.length == 0) {
            return 0;
        }
        for (int num : nums) {
            numSet.add(num);
        }

        for (int num : nums) {
            int tempLongest = 1;
            int currNum = num;
            while (true) {
                if (numSet.contains(currNum + 1)) {
                    tempLongest += 1;
                    currNum += 1;
                } else {
                    break;
                }
            }
            if (tempLongest > longest) {
                longest = tempLongest;
            }
        }
        return longest;
    }
}
